lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -660,6 +659,123 @@ def get(self):
+
+    def clear(self):
+        cmds.checkBoxGrp(self.useOutlinerColor, edit=True, v1=False)
+        cmds.colorSliderGrp(self.outlinerColor, edit=True, rgb=(0,0,0))
+
+    def refresh(self):
+        try:
+            if self.useMetadata:
+                # Use Ufe SceneItem metadata to get values.
+                useOutlinerColor = self.item.getGroupMetadata(self.GROUP, self.USE_OUTLINER_COLOR)
+                assert(not useOutlinerColor.empty())
+                outlinerColor = self.item.getGroupMetadata(self.GROUP, self.OUTLINER_COLOR)
+                cmds.checkBoxGrp(self.useOutlinerColor, edit=True, v1=bool(useOutlinerColor))
+                if not outlinerColor.empty() and (outlinerColor.typeName() == "ufe.Vector3d"):
+                    # Color is stored as double3 USD custom data.
+                    clr = ufe.Vector3d(outlinerColor)
+                    cmds.colorSliderGrp(self.outlinerColor, edit=True,
+                                        rgb=(clr.x(), clr.y(), clr.z()))
+            else:
+                # Get the custom data directly from USD.
+                useOutlinerColor = self.prim.GetCustomDataByKey(self.USE_OUTLINER_COLOR)
+                if useOutlinerColor:
+                    cmds.checkBoxGrp(self.useOutlinerColor, edit=True, v1=useOutlinerColor)
+
+                outlinerColor = self.prim.GetCustomDataByKey(self.OUTLINER_COLOR)
+                if outlinerColor:
+                    # Color is stored as double3 USD custom data.
+                    cmds.colorSliderGrp(self.outlinerColor, edit=True,
+                                        rgb=(outlinerColor[0], outlinerColor[1], outlinerColor[2]))
+        except:
+            self.clear()
+
+    def _updateTextColorChanged(self):
+        '''Update the text color custom data for this prim based on the values
+        set in the two fields.'''
+        currEditTarget = None
+        try:
+            if self.useMetadata:
+                useTextColor = cmds.checkBoxGrp(self.useOutlinerColor, query=True, v1=True)
+                rgb = cmds.colorSliderGrp(self.outlinerColor, query=True, rgbValue=True)
+
+                # Get ufe commands for the two metadata.
+                cmd1 = self.item.setGroupMetadataCmd(self.GROUP, self.USE_OUTLINER_COLOR, useTextColor)
+                ufeVec = ufe.Vector3d(rgb[0], rgb[1], rgb[2])
+                cmd2 = self.item.setGroupMetadataCmd(self.GROUP, self.OUTLINER_COLOR, ufe.Value(ufeVec))
+
+                # Create ufe composite command from both commands above and execute it.
+                cmd = ufe.CompositeUndoableCommand([cmd1, cmd2])
+                ufeCmd.execute(cmd)
+            else:
+                with mayaUsdLib.UsdUndoBlock():
+                    # As initially decided write out the color custom data to the session layer.
+                    stage = self.prim.GetStage()
+                    currEditTarget = stage.GetEditTarget()
+                    stage.SetEditTarget(stage.GetSessionLayer())
```

Comment:
Create undo block ourselves and target sessionlayer directly. Only problem is this won't be edit routable directly, unless we come back later and add it after the other PR is merged.